core/src/main/java/org/zenoss/zep/zing/impl/ZingEventProcessorImpl.java

```diff
@@ -167,9 +173,18 @@ private void processEventSummary(EventSummary summary) {
         }
 
         ZingEvent zingEvent = builder.build();
+
+        PubsubMessage pubsubMessage = this.publisher.getPubSubMessage(zingEvent);
+        int pubsubMessageSize = pubsubMessage.getData().size();
+
         if (zingEvent.isValid()) {
-            logger.debug("publishing event {}", zingEvent);
-            this.publisher.publishEvent(zingEvent);
+            if (pubsubMessageSize >= this.maxPubsubMessageSize) {
+                logger.debug("event {} is too large, dropping", zingEvent.getUuid());
```

Comment:
If we are dropping data that should be sent to Zing, this should be warn at least, not debug.  If we expect this to happen often enough that we don't want the volume of log messages, we should rethink what we are doing.  What is the effect of dropping this data?

Reply:
If ZEP is producing messages that are too big for pubsub - wouldn't it be better to truncate fields rather than drop the whole event?  With this code, we're guaranteed to have events in the CZ that are not in Zing - and it seems like that will be an issue.

Reply:
I agree we should apply some heuristic here to trim the event such that dropping it is more of a hypothetical last resort. I would say before checking for `maxPubsubMessageSize` the code should attempt to truncate a select set of fields.  So we need another configurable value, say `maxFieldLength` which defaults to exactly what I'm not sure - maybe 50K?  That seems pretty large to me. If we need a bigger value, then we can change the config as necessary.

If any of the following fields exceed `maxFieldLength`, log a warning and truncate the field. After inspecting all of the fields, then double check the resultant pubsub message size with the current code.

Here's a strawman for the subset of fields to check against `maxFieldLength` 
```
message
summary
contextTitle
parentContextTitle
details
```

Reply:
Often its trap oids that cause large events for us.  the ones with indexes.  Ideally you would only keep so many traps varbinds with indexes on them.  

Id log that as info so i could by default run with level as warning..debug is tough to go through.

Myvarbind
Myvarbind.1
Myvarbind.1023

Is truncate and not drop..  but also think about the effect on dedup id as well..